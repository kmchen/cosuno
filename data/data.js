export const data =
{
   "company":[
      {
         "name":"AUTO1 Group",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Blacklane",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"Blinkist",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Bonify",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"Camunda",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Cologne"
      },
      {
         "name":"ChartMogul",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
      {
         "name":"Contentful",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["Software"],
         "city": "Cologne"
      },
      {
         "name":"Cosuno",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["construction", "software"],
         "city": "Berlin"
      },
      {
         "name":"Coya",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Crate.io",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Dalia Research",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
      {
         "name":"door2door",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"Eyeo",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Cologne"
      },
      {
         "name":"FreightHub",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"German Autolabs",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"GetYourGuide",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Dusseldorf"
      },
      {
         "name":"GoEuro",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"Helping",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Infarm",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Junique",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"MoBerries",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Morressier",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Movinga",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Frankfurt"
      },
      {
         "name":"N26",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"PEAT",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"PPRO",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"Quentic",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Cologne"
      },
      {
         "name":"ResearchGate",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"Skoove",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Munich"
      },
      {
         "name":"SnapShot",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
      {
         "name":"Watchmaster",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Hamburg"
      },
      {
         "name":"Wundertax",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["fintech", "software"],
         "city": "Berlin"
      },
      {
         "name":"Zeotap",
         "Logo": "https://source.unsplash.com/random",
         "specialty": ["ecommerce", "software"],
         "city": "Munich"
      },
   ]
};
