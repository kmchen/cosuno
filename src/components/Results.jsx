import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';

const useStyles = makeStyles(theme => ({
  cardGrid: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  cardMedia: {
    paddingTop: '56.25%', // 16:9
  },
  cardContent: {
    flexGrow: 1,
  },
}));

export default function Results({ selectedCompany }) {
  const classes = useStyles();
  return (
    <Container className={classes.cardGrid} maxWidth="sm">
      <Grid container spacing={4}>
        {selectedCompany.map(company => (
          <Grid item key={company.name} xs={12} sm={6} md={4}>
            <Card className={classes.card}>
              <CardMedia
                className={classes.cardMedia}
                image="https://source.unsplash.com/random"
                title="Image title"
              />
              <CardContent className={classes.cardContent}>
                <Typography gutterBottom variant="h5" component="h2">
                  {company.name}
                </Typography>
                <Typography>
                  City:
                  {company.city}
                </Typography>
                <Typography>
                  Specialty:
                  {company.specialty.join(', ')}
                </Typography>
              </CardContent>
            </Card>
          </Grid>
        ))}
      </Grid>
    </Container>
  );
}

/* eslint-disable react/forbid-prop-types */
Results.propTypes = {
  selectedCompany: PropTypes.array.isRequired,
};
