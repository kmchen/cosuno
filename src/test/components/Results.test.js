import React from 'react';
import { shallow, configure } from 'enzyme';
import Fuse from 'fuse.js';
import Adapter from 'enzyme-adapter-react-16';
import Results from '../../components/Results';
import { data } from '../../../data/data.js';

configure({ adapter: new Adapter() });

describe('Results component', () => {
  let ResultsComponent;

  beforeEach(() => {
    const props = {
      selectedCompany: [{
        name:"AUTO1 Group",
        Logo: "https://source.unsplash.com/random",
        specialty: ["ecommerce", "software"],
        city: "Hamburg"
      }]
    };
    ResultsComponent = shallow(<Results {...props} />);
  });

  it('should render Results component', () => {
    expect(ResultsComponent).toMatchSnapshot();
  });

  it('should render Card component', () => {
    const cardComponent = ResultsComponent.find('WithStyles(ForwardRef(Card))');
    expect(cardComponent.exists()).toBeTruthy();
  });
});

