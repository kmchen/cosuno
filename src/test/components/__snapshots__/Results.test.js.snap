// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component should render Results component 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-cardGrid-1"
  maxWidth="sm"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="AUTO1 Group"
      md={4}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Card))
        className="makeStyles-card-2"
      >
        <WithStyles(ForwardRef(CardMedia))
          className="makeStyles-cardMedia-3"
          image="https://source.unsplash.com/random"
          title="Image title"
        />
        <WithStyles(ForwardRef(CardContent))
          className="makeStyles-cardContent-4"
        >
          <WithStyles(ForwardRef(Typography))
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            AUTO1 Group
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))>
            City:
            Hamburg
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))>
            Specialty:
            ecommerce, software
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
